fedimint-core/src/module/mod.rs

```diff
@@ -874,7 +874,12 @@ pub trait ServerModule: Debug + Sized {
     ///
     /// Summing over all modules, if liabilities > assets then an error has
     /// occurred in the database and consensus should halt.
-    async fn audit(&self, dbtx: &mut ModuleDatabaseTransaction<'_>, audit: &mut Audit);
+    async fn audit(
+        &self,
+        dbtx: &mut ModuleDatabaseTransaction<'_>,
+        audit: &mut Audit,
+        module_instance_id: ModuleInstanceId,
+    );
```

Comment:
We'll have to re-think this API, ideally modules aren't aware of their own instance id. We might want to remove the DB as a communication mechanism and let modules do a tally internally and directly return the audit items? Thoughts @dpc?

Reply:
Agreed.